### ISSUE TITLE
Image normalization

### ISSUE THREAD

**Issue by cglasner** [MSKCC-Computational-Pathology/MIL-nature-medicine-2019#9]
Hello everyone!

I have a question regarding the preprocessing of the images that go into the CNN portion of this system. The CNN starts out as a pretrained resnet34 from torchvision.pytorch. [Pytorch docs for these models](https://pytorch.org/docs/stable/torchvision/models.html) provide the following info:

> The images have to be loaded in to a range of [0, 1] and then normalized using mean = [0.485, 0.456, 0.406] and std = [0.229, 0.224, 0.225].

Literally while writing this issue I figured out that the range thing is being achieved via [transforms.ToTensor](https://pytorch.org/docs/stable/torchvision/transforms.html#torchvision.transforms.ToTensor) (just in case anybody else reading this issue was wondering about that, it's fine). However I'm still a little confused about the normalization. In the code, mean = [0.5, 0.5, 0.5] and std = [0.1, 0.1, 0.1] are used instead of the values proposed by the docs. Is there some special reason for this? While researching this, I did find some people claiming that when your training data is very different from ImageNet data (which is certainly the case here), then using the mean and std of your own training data instead might be worthwhile. Is this what happened here? I would appreciate some clarification on the matter. :-)

Thank you and have a nice holiday!

**Comment by gabricampanella** [comment]
It is common to calculate mean and sd of each channel for your dataset and use those values to normalize the data. While the network could be able to learn the normalization implicitly if necessary, empirically it is known that normalizing does help training. We used values of 0.5 and 0.1 for every channel because in our experience there is no evident gain when using the true mean and sd for WSI slides.

**Comment by cglasner** [comment]
Thank you for the answer!
